### ISSUE TITLE
Using a custom OAuth Client

### ISSUE THREAD

**Issue by sboardwell** [google/clasp#901]
> (Note: Non-breaking issues are likely not to be prioritized. Please consider a PR in addition to your issue)

## Question

With respect to #85, and in particular [this comment](https://github.com/google/clasp/issues/85#issuecomment-721398154) would it make sense to make the clientID and secret configurable?

The idea would be to allow corporations to create their own **trusted** client and pass the clientId and clientSecret to the clasp command, either through parameters or the `.clasprc.json` file.

Would it be possible to get the steps necessary to create the client (the list of permissions)?

I would be happy to create a PR if you point me in the right direction.

**Comment by sqrrrl** [comment]
Yep, this is coming in the next major version and will be the recommended approach.

**Comment by sqrrrl** [comment]
3.0 alpha available and has support for bring-your-own-client. Readme updated with instructions on how to set up.
